Lab2/src/main/java/ua/lviv/IoT/lab2/manager/impl/CheckWord.java

```diff
@@ -0,0 +1,30 @@
+package ua.lviv.IoT.lab2.manager.impl;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+
+public class CheckWord {
+   static public Set<String> checkWord(final String input, int n) {
+      String[] listOfWord = input.toLowerCase().split("\\W");
+      Set<String> listOfResult = new HashSet<>();
+      int k = 1;
+      int j = 1;
+      for (String word : listOfWord) {
+         for (int i = j; i < listOfWord.length; i++) {
+            if (!word.equals("")) {
+               if (listOfWord[i].equals(word)) {
+                  k++;
+               }
+            }
+         }
+         if (k > n) {
+            listOfResult.add(word);
+         }
+         k = 1;
+         j++;
+      }
+      return listOfResult;
```

Comment:
I have a suggestion for you: why not make a set of unique words from the listOfWord, and then use a HashMap where the key is a word and the value is the number of occurrences in the text?
And then you simply traverse through the listOfWord and add 1 to counter for this word.
Then you return all the keys where value is greater than `n`.
This way, you don't have a for loop inside of another for loop, and this should be faster.

But it's not mandatory, we can just discuss your solutions and better alternatives during our time slot.

Possible references: [HashMap](https://www.w3schools.com/java/java_hashmap.asp), [Convert Set to Map](https://www.techiedelight.com/convert-set-to-map-java)